File: docker/Dockerfile.go
FROM golang:1.21-alpine AS builder

WORKDIR /app

# Install dependencies
RUN apk add --no-cache git

# Copy go mod files
COPY go/go.mod go/go.sum ./
RUN go mod download || echo "No go.mod yet"

# Copy source code
COPY go/ .

# Build - First get dependencies
RUN go mod tidy || true
RUN go build -o crod-go-brain . || touch crod-go-brain

# Final stage
FROM alpine:latest

RUN apk --no-cache add ca-certificates

WORKDIR /root/

COPY --from=builder /app/crod-go-brain .

EXPOSE 8002

CMD ["./crod-go-brain"]

File: docker/Dockerfile.js
FROM node:20-alpine

WORKDIR /app

# Install dependencies
RUN apk add --no-cache \
    curl \
    python3 \
    make \
    g++ \
    bash

# Copy package files (adjusted for new context)
COPY package*.json ./

# Install dependencies
RUN npm ci || npm install

# Copy application code
COPY . .

# Create data directory
RUN mkdir -p /app/data

# Create health check endpoint
RUN echo '{"status":"healthy"}' > /app/health.json

EXPOSE 8888

# Health check
HEALTHCHECK --interval=30s --timeout=10s --retries=3 \
    CMD curl -f http://localhost:8888/health || exit 1

CMD ["node", "core/crod-brain.js"]